### DISCUSSION TITLE
Question on PB via cPanel hosting

### DISCUSSION THREAD

**Discussion by Karlstens** [pocketbase/pocketbase#5150] [Q&A] [answered]
Learning about pocketbase for the first time this week, I was impressed by how easy it was to get loaded onto my Windows Desktop, running via Node.JS/PM2 to keep it as a background process - allowing me to learn and tinker locally on my desktop. Very cool! 

The next stage I have with my learning is how I could host pocketbase on a subdomain on my website - a shared server via cPanel. Any general guidance and steps that will assist with getting an off-the-shelf pocketbase instance up and running would be greatly appreciated.

Am I simply to follow the "Minimal setup" steps found under Deployment Strategies in the [Going to production](https://pocketbase.io/docs/going-to-production/) articles? Or will there be more steps to bringing PB online within a Shared Server cPanel environment?

**Comment by ganigeorgiev** [comment, marked as answer]
It is not possible to use PocketBase on a shared hosting because you'll need to able to execute custom binary and bind/open custom ports and most shared hosting providers that I've used don't allow this.

You need a VPS or dedicated server to host PocketBase and you'll probably stumble on the same if you try to host anything different than PHP on a sandboxed shared hosting platform.

**Reply by Karlstens** [reply]
Thanks for that advice, much appreciated.

**Reply by Swe-HimelRana** [reply]
So how expressjs and python app run in cpanel?

**Reply by ganigeorgiev** [reply]
@Swe-HimelRana I have no idea and it depends on the provider and what type of virtualization/isolation they have implemented. Most of the shared hosting providers that I've used doesn't have such options or are limited to cgi type of execution. Contact your shared hosting provider and ask them.
